stage1/init/common/pod.go

```diff
@@ -302,62 +302,13 @@ func appToSystemd(p *stage1commontypes.Pod, ra *schema.RuntimeApp, interactive b
 		return errwrap.Wrap(errors.New("unable to write environment file"), err)
 	}
 
-	var _uid, gid int
-	var err error
-
-	uidRange := uid.NewBlankUidRange()
-	if err := uidRange.Deserialize([]byte(privateUsers)); err != nil {
-		return errwrap.Wrap(errors.New("unable to deserialize uid range"), err)
-	}
-
-	if strings.HasPrefix(app.User, "/") {
-		var stat syscall.Stat_t
-		if err = syscall.Lstat(filepath.Join(common.AppRootfsPath(p.Root, appName),
-			app.User), &stat); err != nil {
-			return errwrap.Wrap(fmt.Errorf("unable to get uid from file %q",
-				app.User), err)
-		}
-		uidReal, _, err := uidRange.UnshiftRange(stat.Uid, 0)
-		if err != nil {
-			return errwrap.Wrap(errors.New("unable to determine real uid"), err)
-		}
-		_uid = int(uidReal)
-	} else {
-		_uid, err = strconv.Atoi(app.User)
-		if err != nil {
-			_uid, err = passwd.LookupUidFromFile(app.User,
-				filepath.Join(common.AppRootfsPath(p.Root, appName), "etc/passwd"))
-			if err != nil {
-				return errwrap.Wrap(fmt.Errorf("cannot lookup user %q", app.User), err)
-			}
-		}
-	}
-
-	if strings.HasPrefix(app.Group, "/") {
-		var stat syscall.Stat_t
-		if err = syscall.Lstat(filepath.Join(common.AppRootfsPath(p.Root, appName),
-			app.Group), &stat); err != nil {
-			return errwrap.Wrap(fmt.Errorf("unable to get gid from file %q",
-				app.Group), err)
-		}
-		_, gidReal, err := uidRange.UnshiftRange(0, stat.Gid)
-		if err != nil {
-			return errwrap.Wrap(errors.New("unable to determine real gid"), err)
-		}
-		gid = int(gidReal)
-	} else {
-		gid, err = strconv.Atoi(app.Group)
-		if err != nil {
-			gid, err = group.LookupGidFromFile(app.Group,
-				filepath.Join(common.AppRootfsPath(p.Root, appName), "etc/group"))
-			if err != nil {
-				return errwrap.Wrap(fmt.Errorf("cannot lookup group %q", app.Group), err)
-			}
-		}
+	Uid, Gid, err := parseUserGroup(p, ra, privateUsers)
```

Comment:
just curious, does Golang advise an upper-case first letter to avoid conflicts with package name, or did you just decide it?